### ISSUE TITLE
"clean" on osx projects always cleans the openframeworks lib (xcode 6.3.2) 

### ISSUE THREAD

**Issue by ofZach** [openframeworks/openFrameworks#4560]
on xcode 6.3.2, osx 10.11.1 

clean seems to delete openframeworks lib as well as the project -- there's not easy way I can see to just clean for the scheme (ie, just the src of the project).  

this seems to have always been the case, but now it's much more painful because we are compiling for 32 bit and 64 bit at once, it feels like it takes forever if you need to clean a project.    when updating 0.8.4 projects to 0.9.0 it seems to help to clean them...    (in this case, the older .app is conflicting a bit with rsync, etc) 

this is probably part of a larger conversation about a better way to include the OF lib as a dependency -- it feels like there might be a better way to do this?

**Comment by ofTheo** [comment]
In older Xcode versions it would pop up an option to clean just the project or both the project and OF, but I believe Xcode got rid of that feature. Now it just cleans everything. 

I think soon we could get to a point where OF is included as a lib and not a project dependency.

**Comment by ofZach** [comment]
yeah I think this move would be really helpful -- I remember that pop up, and really start to miss it now -- when I do clean it makes me feel like I'm on an rPi :)

**Comment by HalfdanJ** [comment]
`I think soon we could get to a point where OF is included as a lib and not a project dependency.`
Could be solved in project generator? Having different templates, one for projects where you dont need to touch core (should be most projects), and one for oF developers that want easily to change stuff

**Comment by ofZach** [comment]
yes that's totally possible down the line...   right now the gui doesn't really expose templates as an option, but if folks get into the command line, the template feature can be really helpful for making new projects act a certain way.  Right now we don't have great support in the PG for templates and "recursive update" -- that's something we've been discussing (having some metadata similar to addons.make that would help specify the template).  I know @kylemcdonald was interested in this for his minimal main.cpp only examples.

one worry I have about multiple templates that are xcproj specific is that it becomes a bit of work to keep them all up to date if we change settings, etc.
